addons/t9n/views/t9n_resource_views.xml

```diff
@@ -0,0 +1,31 @@
+<?xml version="1.0"?>
+<odoo>
+    <record id="t9n.resource_view_form" model="ir.ui.view">
+        <field name="name">Resource</field>
+        <field name="model">t9n.resource</field>
+        <field name="arch" type="xml">
+            <form string="Resource Form">
+                <sheet>
+                    <group>
+                        <group>
+                            <field name="file_name" readonly="True"/>
+                        </group>
+                        <group>
+                            <field name="file" filename="file_name"/>
+                        </group>
+                    </group>
+                    <field name="message_ids"/>
+                </sheet>
+            </form>
+        </field>
+    </record>
+    <record id="t9n.resource_view_tree" model="ir.ui.view">
+        <field name="name">t9n.resource.tree</field>
+        <field name="model">t9n.resource</field>
+        <field name="arch" type="xml">
+            <tree>
```

Comment:
if we're not going to allow directly creatung a resource in the tree view for now, you can add in an attribute to prevent users from being able to:

```suggestion
            <tree create="0">
```

Reply:
We need to be able to create a resource from the tree view of the project.